### PR TITLE
Remove remaining uses of Instruction-constructors that insert before another Instruction

### PR DESCRIPTION
**Pull request by SLTozer**
Related to the work noted in [discourse](https://discourse.llvm.org/t/psa-instruction-constructors-changing-to-iterator-only-insertion/77845) and in [this](https://github.com/llvm/llvm-project/pull/85980) patch.

This removes the remaining set of callsites where instructions are created using constructors that take a `Instruction *InsertBefore` argument, in preparation for those constructors being removed. For an explanation on why we want to do this, see the discussions above; the short summary is that it is useful for ensuring that debug info placement is handled correctly, preventing potential errors from emerging following the RemoveDIs work. In almost all cases this should be NFC, except for a few cases where `getFirstNonPHI()` is replaced by `getFirstNonPHIIt()`, which may prevent a possible error from occurring.

### CODE REVIEW

File: llvm/include/llvm/IR/InstrTypes.h [hunk 1]
```diff
@@ -1542,19 +1542,10 @@ class CallBase : public Instruction {
...
-                                    OperandBundleDef OB,
-                                    BasicBlock::iterator InsertPt);
-
   /// Create a clone of \p CB with operand bundle \p ID removed.
   static CallBase *removeOperandBundle(CallBase *CB, uint32_t ID,
                                        Instruction *InsertPt = nullptr);
 
-  /// Create a clone of \p CB with operand bundle \p ID removed.
-  static CallBase *removeOperandBundle(CallBase *CB, uint32_t ID,
-                                       BasicBlock::iterator InsertPt);
```
Comment (OCHyams) [inline llvm/include/llvm/IR/InstrTypes.h:1556]: why are these removed?

Comment (SLTozer) [reply llvm/include/llvm/IR/InstrTypes.h:1556]: Looks like a change that accidentally snuck in while doing some git stuff, will reinstate.

File: llvm/lib/FuzzMutate/IRMutator.cpp [hunk 2]
```diff
@@ -542,7 +542,8 @@ void InsertPHIStrategy::mutate(BasicBlock &BB, RandomIRBuilder &IB) {
   if (&BB == &BB.getParent()->getEntryBlock())
     return;
   Type *Ty = IB.randomType();
-  PHINode *PHI = PHINode::Create(Ty, llvm::pred_size(&BB), "", &BB.front());
+  PHINode *PHI =
+      PHINode::Create(Ty, llvm::pred_size(&BB), "", BB.getFirstInsertionPt());
```
Comment (OCHyams) [inline llvm/lib/FuzzMutate/IRMutator.cpp:546]: `BB.begin()` instead of `BB.getFirstInsertionPt()`?

**Review by OCHyams — APPROVED** [review]
LGTM if this is clang-formatted (I can't tell). Plus one small inline nit - I think it's fine either way.
